spec/schemas/_common.yaml

```diff
@@ -526,18 +549,9 @@ components:
               required:
                 - source
     ScriptLanguage:
-      anyOf:
-        - title: builtin
-          $ref: '#/components/schemas/BuiltinScriptLanguage'
-        - title: custom
-          type: string
-    BuiltinScriptLanguage:
       type: string
-      enum:
-        - expression
-        - java
-        - mustache
-        - painless
+      description: |-
+        Built in scripts include: 'expression', 'java', 'mustache', and 'painless', but custom script languages can be provided as well.
```

Comment:
I'm not sure I agree with removing information from the specification. I feel like if you want these to be collapsed/simplified, that could be handled at a tooling/translation level?

Additionally these have generated counterparts that are used in opensearch-java:
- https://github.com/opensearch-project/opensearch-java/blob/main/java-client/src/generated/java/org/opensearch/client/opensearch/_types/BuiltinScriptLanguage.java
- https://github.com/opensearch-project/opensearch-java/blob/main/java-client/src/generated/java/org/opensearch/client/opensearch/_types/ScriptLanguage.java

Reply:
I'm not sure why `anyOf` represents a problem or inaccuracy? It is true that any known/built-in script language is also a valid "custom" language, but not all "custom" languages will be "known" languages. 

If you really wanted this to be a `oneOf`, you could achieve it via:
```yaml
    ScriptLanguage:
      oneOf:
        - title: builtin
          $ref: '#/components/schemas/BuiltinScriptLanguage'
        - title: custom
          type: string
          not:
            $ref: '#/components/schemas/BuiltinScriptLanguage'
```

The primary goal of the specification at present is enabling the generation of the language clients, not necessarily to match the server's internal representation. So being able to provide an enum of core/commonly used values is helpful to end users.